edx_notifications/stores/sql/tests/test_store_provider.py

```diff
@@ -578,6 +578,43 @@ def test_get_notifications_for_user(self):
             )
             self.assertEqual(len(notifications), 0)
 
+        #
+        # test start_date and end_date filtering.
+        #
+        self.assertEqual(
+            self.provider.get_num_notifications_for_user(
+                self.test_user_id,
+                filters={
+                    'start_date': msg1.created.date(),
+                    'end_date': msg2.created.date()
+                }
+            ),
+            2
+        )
+
+        with self.assertNumQueries(1):
+            notifications = self.provider.get_notifications_for_user(
+                self.test_user_id,
+                filters={
+                    'start_date': msg1.created.date(),
+                    'end_date': msg2.created.date()
+                }
+            )
+            self.assertEqual(len(notifications), 2)
+            self.assertEqual(notifications[0].msg, msg2)
+            self.assertEqual(notifications[1].msg, msg1)
+
+        self.assertEqual(
```

Comment:
Can we write a test where msg2's create date is updated and then we write a query so that we only get msg1 back?